sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RewriteSubquerySuite.scala

```diff
@@ -33,23 +34,44 @@ class RewriteSubquerySuite extends PlanTest {
       Batch("Rewrite Subquery", FixedPoint(1),
         RewritePredicateSubquery,
         ColumnPruning,
+        InferFiltersFromConstraints,
+        PushDownPredicate,
         CollapseProject,
         RemoveRedundantProject) :: Nil
   }
 
   test("Column pruning after rewriting predicate subquery") {
-    val relation = LocalRelation('a.int, 'b.int)
-    val relInSubquery = LocalRelation('x.int, 'y.int, 'z.int)
+    withSQLConf(SQLConf.CONSTRAINT_PROPAGATION_ENABLED.key -> "false") {
+      val relation = LocalRelation('a.int, 'b.int)
+      val relInSubquery = LocalRelation('x.int, 'y.int, 'z.int)
 
-    val query = relation.where('a.in(ListQuery(relInSubquery.select('x)))).select('a)
+      val query = relation.where('a.in(ListQuery(relInSubquery.select('x)))).select('a)
 
-    val optimized = Optimize.execute(query.analyze)
-    val correctAnswer = relation
-      .select('a)
-      .join(relInSubquery.select('x), LeftSemi, Some('a === 'x))
-      .analyze
+      val optimized = Optimize.execute(query.analyze)
+      val correctAnswer = relation
+        .select('a)
+        .join(relInSubquery.select('x), LeftSemi, Some('a === 'x))
+        .analyze
 
-    comparePlans(optimized, correctAnswer)
+      comparePlans(optimized, correctAnswer)
+    }
+  }
+
+  test("Infer filters and push down predicate after rewriting predicate subquery") {
```

Comment:
Need the column pruning in the test title?

Reply:
How about making the test title simple, then leaving comments about what's tested clearly here?

Reply:
How about refactor these test to:
```scala
  val relation = LocalRelation('a.int, 'b.int)
  val relInSubquery = LocalRelation('x.int, 'y.int, 'z.int)

  test("Column pruning") {
    withSQLConf(SQLConf.CONSTRAINT_PROPAGATION_ENABLED.key -> "false") {
      val query = relation.where('a.in(ListQuery(relInSubquery.select('x)))).select('a)

      val optimized = Optimize.execute(query.analyze)
      val correctAnswer = relation
        .select('a)
        .join(relInSubquery.select('x), LeftSemi, Some('a === 'x))
        .analyze

      comparePlans(optimized, correctAnswer)
    }
  }

  test("Column pruning, infer filters and push down predicate") {
    withSQLConf(SQLConf.CONSTRAINT_PROPAGATION_ENABLED.key -> "true") {
      val query = relation.where('a.in(ListQuery(relInSubquery.select('x)))).select('a)

      val optimized = Optimize.execute(query.analyze)
      val correctAnswer = relation
        .where(IsNotNull('a)).select('a)
        .join(relInSubquery.where(IsNotNull('x)).select('x), LeftSemi, Some('a === 'x))
        .analyze

      comparePlans(optimized, correctAnswer)
    }
  }
```